Source/Core/Core.js

```diff
@@ -57,27 +57,32 @@ var instanceOf = this.instanceOf = function(item, object){
 	return item instanceof object;
 };
 
-// Function overloading
-
-var Function = this.Function;
-
+/*<ltIE8>*/
 var enumerables = true;
 for (var i in {toString: 1}) enumerables = null;
 if (enumerables) enumerables = ['hasOwnProperty', 'valueOf', 'isPrototypeOf', 'propertyIsEnumerable', 'toLocaleString', 'toString', 'constructor'];
+/*</ltIE8>*/
+var hasOwnProperty = Object.prototype.hasOwnProperty;
+function eachKey(object, fn, thisArg){
```

Comment:
I think `eachKey` isn't a very nice name. maybe `objectEach`, or just `each`?

Also I'm thinking if this shouldn't be:

```js
var objectKeys = Object.keys || function(object){
   // ...
   return keys;
};
```

And use that in the overloadSetter, with a for-loop there.

And below, just have:

```js
Object.extend({
  keys: objectKeys,
  forEach: // ...
});
```